pkg/destroy/aws/iamhelpers.go

```diff
@@ -39,6 +41,13 @@ func (search *iamRoleSearch) find(ctx context.Context) (arns []string, names []s
 				// Unfortunately role.Tags is empty from ListRoles, so we need to query each one
 				response, err := search.client.GetRoleWithContext(ctx, &iam.GetRoleInput{RoleName: role.RoleName})
 				if err != nil {
+					if quotaaws.IsUnauthorized(err) {
+						// Installer does not have access to this IAM role
+						// Ignore this IAM Role and donot report this error via
+						// lastError
+						search.unmatched[*role.Arn] = exists
+						continue
+					}
```

Comment:
It looks a bit weird to import a `quota` package to check if an error is `AccessDenied`. Either `IsUnauthorized` should be moved out of the quota package so it can be shared with multiple packages, or we should just have our own `isUnauthorized` function in this file. Actually, I think this whole block would look better with a `switch`

```suggestion
                    var awsErr awserr.Error
                    if errors.As(err, &awsErr) {
                        switch awsErr.Code() {
                        case "AccessDeniedException":
						    // Installer does not have access to this IAM role
						    // Ignore this IAM Role and do not report this error via
						    // lastError
						    search.logger.Debugf("AccessDenied to role %s. Expected if this is an STS install", *role.Arn)
						    fallthrough
					    case iam.ErrCodeNoSuchEntityException:
						    search.unmatched[*role.Arn] = exists
						    continue
						default:
						}
			        }
					if lastError != nil {
					    search.logger.Debug(lastError)
					}
					lastError = errors.Wrapf(err, "get tags for %s", *role.Arn)
				}
```

Reply:
It would be nice to have one helper function which identifies unauthorized access for any AWS API's return code and that was the reason behind using the pre-existing method (unfortunately it exists in quota/aws).  
Yes, this lends itself to using a switch statement.  
I am concerned if logging a statement for the AccessDenied case would make the logs noisy hence stayed away from it. Happy to add it if it provides value to the customer (in non STS cases maybe?)